sys/arm64/arm64/db_trace.c

```diff
@@ -94,6 +94,14 @@ db_stack_trace_cmd(struct thread *td, struct unwind_state *frame)
 			struct trapframe *tf;
 
 			tf = (struct trapframe *)(uintptr_t)frame->fp - 1;
+#ifdef __CHERI_PURE_CAPABILITY__
+			if (!cheri_can_access(tf,
+			    CHERI_PERM_LOAD | CHERI_PERM_LOAD_CAP,
+			    (vm_offset_t)tf, sizeof(*tf))) {
+				db_printf("--- invalid trapframe %#p\n", tf);
+				break;
+			}
+#endif
 			if (!kstack_contains(td, (vm_offset_t)tf,
```

Comment:
I wonder if we should be checking alignment too upstream and downstream like we do in unwind_frame?

Reply:
Yes, I think we should add the alignment check upstream.